core/blockchain.go

```diff
@@ -1459,6 +1459,20 @@ func (bc *BlockChain) reorgNeeded(localBlock *types.Block, localTd *big.Int, ext
 	return reorg
 }
 
+// pruneBlockSidecars prunes the sidecars of blocks that are older than the keep period
+func (bc *BlockChain) pruneBlockSidecars(db ethdb.KeyValueWriter, curBlock *types.Block) {
+	if curBlock.NumberU64() < uint64(params.BlobPrunePeriod) {
+		return
+	}
+	pruneBlockNumber := curBlock.NumberU64() - uint64(params.BlobPrunePeriod)
+	pruneBlock := bc.GetBlockByNumber(pruneBlockNumber)
+	if pruneBlock == nil {
+		return
+	}
+	pruneBlockHash := pruneBlock.Hash()
```

Comment:
I'm not sure if it is better but we can use bc.GetCanonicalHash to get the block hash

Reply:
It is cleaner I believe, fixed.